_studio/shared/umc/codec/h264_dec/include/umc_h264_task_supplier.h

```diff
@@ -788,19 +785,19 @@ inline int32_t CalculateDPBSize(uint8_t & level_idc, int32_t width, int32_t heig
             level_idc = H264VideoDecoderParams::H264_LEVEL_51;
             break;
 
-        // Set level_idc to INTERNAL_MAX_LEVEL so that MaxDPBMbs = 696320
-        // can be used to calculate the DPB size.
-        case H264VideoDecoderParams::H264_LEVEL_51:
-        case H264VideoDecoderParams::H264_LEVEL_52:
-            level_idc = H264VideoDecoderParams::H264_LEVEL_52;
-            break;
 #if (MFX_VERSION >= MFX_VERSION_NEXT)
         case H264VideoDecoderParams::H264_LEVEL_6:
         case H264VideoDecoderParams::H264_LEVEL_61:
         case H264VideoDecoderParams::H264_LEVEL_62:
-            level_idc = H264VideoDecoderParams::H264_LEVEL_MAX;
-            break;
+            // fall-through
 #endif
+        // Set level_idc to INTERNAL_MAX_LEVEL so that MaxDPBMbs = 696320
+        // can be used to calculate the DPB size.
+        case H264VideoDecoderParams::H264_LEVEL_51:
+        case H264VideoDecoderParams::H264_LEVEL_52:
```

Comment:
If I got right this change was a root cause of the regression from https://github.com/Intel-Media-SDK/MediaSDK/commit/e126273431b470a738e4c26896b829a6801cb455:
```
        case H264VideoDecoderParams::H264_LEVEL_52:
-            level_idc = INTERNAL_MAX_LEVEL;
+           level_idc = H264VideoDecoderParams::H264_LEVEL_52;
```
and you restored it back. 
Thank you @uartie !

Reply:
right.

Reply:
I think I will move these cases back up so they remain in increasing order.

Reply:
updated.